### DISCUSSION TITLE
High delay when iterating over JavaScript async iterator in Python

### DISCUSSION THREAD

**Discussion by CNSeniorious000** [pyodide/pyodide#4492] [Q&A]
If my laptop is not plugged in, there will be a significant delay when iterating over a JavaScript async iterator in Python.

I am iterating a JavaScript async iterator in Python like #4360, and found that it is slow. But when I plugged in, it is significantly faster.

I tried unplugged and plugged several times, and found that reproduceable.

**Comment by ryanking13** [comment]
There might be some logic in the laptop that optimizes battery time, by trading off the CPU performance when not unplugged. This sounds interesting but I guess there is nothing that Pyodide can do about this...?

**Reply by CNSeniorious000** [reply]
That's possible, but in fact the iterator is doing nothing with the CPU, it is iterating a streaming HTTP response. Does any HTTP API's performance is limited when optimizing battery time?

Sorry for not providing a minimal reproduceable example. I may make one if I have time.

> I guess there is nothing that Pyodide can do about this...?

I think this is right, so I am closing this. Thank you~
